File: food-delivery-app-backend/go/payment-service/internal/application/dto/payment_event_payload.go
package dto

import (
	"food-delivery-app-backend/payment-service/internal/domain/valueobject"
	"time"
)

type PaymentEventPayload struct {
	EventId       string
	PaymentId     string
	OrderId       string
	CustomerId    string
	WalletId      string
	Amount        float64
	PaymentStatus valueobject.PaymentStatus
	CreatedAt     time.Time
}

File: food-delivery-app-backend/java/order-service/order-command/src/main/java/com/food/delivery/app/order/command/features/process_order_payments/v1/dto/PaymentEventPayload.java
package com.food.delivery.app.order.command.features.process_order_payments.v1.dto;

import com.food.delivery.app.order.command.domain.valueobjects.PaymentStatus;
import lombok.Builder;
import lombok.Getter;

import java.sql.Timestamp;
import java.util.UUID;

@Getter
@Builder
public class PaymentEventPayload {
    private UUID eventId;
    private UUID paymentId;
    private UUID orderId;
    private UUID customerId;
    private UUID walletId;
    private double amount;
    private PaymentStatus paymentStatus;
    private Timestamp createdAt;
    private String errorMessage;
}
